src/curvefit/solvers/solvers.py

```diff
@@ -99,4 +100,32 @@ def fit(self, data, x_init=None, options=None):
                 xs_opt.append(self.solver.x_opt)
             
             self.x_opt = xs_opt[np.argmin(fun_vals)]
-            self.fun_val_opt = np.min(fun_vals)
\ No newline at end of file
+            self.fun_val_opt = np.min(fun_vals)
+
+
+class SmartInitialization(Composite):
+
+    def __init__(self, update_prior_fun):
+        self.update_prior_fun = update_prior_fun
+
+    def fit(self, data, x_init=None, options=None):
+         if self.is_solver_defined():
+            model = self.get_model_instance()
+            if x_init is None:
+                x_init = model.x_init
+            df = data[0]
+            data_specs = data[1]
+            group_names = df[data_specs.col_group].unique()
+            xs = []
+            for group in group_names:
+                data_sub = (df[df[data_specs.col_group] == group], data_specs)
+                self.solver.fit(data_sub, x_init, options)
+                xs.append(self.solver.x_opt)
+                model.erase_data()
+
+            new_param_set = self.update_prior_fun(model.param_set, xs)
+            model.update_param_set(new_param_set)
```

Comment:
Aha! Is see why you thought that maybe PR #79 supersedes this... I think this actually works fine as itself because it's initialization inside of a model rather than outside of a model. For example, I would actually use this `SmartInitialization` inside of the `PriorInitialization`.